src/algebraic_geometry/presheafed_space.lean

```diff
@@ -0,0 +1,173 @@
+-- Copyright (c) 2019 Scott Morrison. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Scott Morrison
+import category_theory.instances.Top.presheaf
+
+universes v u
+
+open category_theory
+open category_theory.instances
+open category_theory.instances.Top
+open topological_space
+
+variables (C : Type u) [𝒞 : category.{v+1} C]
+include 𝒞
+
+namespace algebraic_geometry
+
+structure PresheafedSpace :=
+(X : Top.{v})
+(𝒪 : X.presheaf C)
+
+variables {C}
+
+namespace PresheafedSpace
+
+instance : has_coe_to_sort (PresheafedSpace.{v} C) :=
+{ S := Type v, coe := λ F, F.X.α }
+
+instance (F : PresheafedSpace.{v} C) : topological_space F := F.X.str
```

Comment:
Nitpicking about notation... I'm inclined to denote a term of type `PresheafedSpace C` with `X` (or `Y`, etc) and think of it as the space (equipped with a presheaf). We could then have
```
def structure_presheaf (X) : presheaf C X := X.𝒪
```
The notation `F` reminds me too much of a (pre)sheaf, which leads to cognitive disconance on lines like this. (There is no topological_space structure on the presheaf... what the hack is going on... oooh... it's the underlying space.)

Reply:
I agree. I think the only obstacle is what to call the underlying `Top`. If we leave that field name as `X`, then we find ourselves writing `X.X`, which is a bit confusing.

We could rename the field as `to_Top`, I guess.

Reply:
This looks nice to me.